nixos/lib/test-driver/test-driver.py

```diff
@@ -566,6 +567,42 @@ def screenshot(self, filename: str) -> None:
             if ret.returncode != 0:
                 raise Exception("Cannot convert screenshot")
 
+    def copy_from_host_via_shell(self, source: str, target: str) -> None:
```

Comment:
Is there a reason we have this?

Looking at https://github.com/NixOS/nixpkgs/pull/78670/commits/510ada705e9f860a011a74394b0b2deed304a440, these could probably also just use `copy_from_host`…

Reply:
At first i am using `copy_from_host`, but with the machine configs that follow later in the script, it doesn't work any longer as the VMs do not seem to support this. For that reason i added the shell variant as an addition.

Reply:
I assume this is because `/tmp/shared` from `nixos/modules/virtualisation/qemu-vm.nix` isn't setup in the installer.

Can we just invoke the `mount` command manually before running `copy_from_host`?

Reply:
We could, but i really don't like the implications of this: Currently, tests are agnostic of how this is all implemented and this allows for transparently changing the virt backend etc.

As soon as tests begin doing such things, they would become testdriver-implementation specific and i am pretty sure we don't want that. I'd rather have two different functions that communicate which preconditions they assume and let the user choose from them, before users hard-code testdriver-details into their tests.

Reply:
We had an offline discussion on that. Agreed on not leaking implementation-specific details about how things are shared into the actual test code. Ideally, this should be encapsulated behind the corresponding methods.